src/type_set.rs

```diff
@@ -84,9 +83,9 @@ where
     pub(crate) fn create_component_write_systems(
         &self,
         dispatcher: &mut DispatcherBuilder,
-        deserializer_map: &mut ComponentDeserializerMap,
+        map: ComponentMap,
```

Comment:
How does this work if the map is being passed in by value? The reason it was being passed in as a `&mut` before was so that the call to `create_component_write_systems` could populate the map with the channels as the corresponding systems are created. If the map is passed in by value, then the calling code won't be able to see any modifications that are made to the map within the function.

Reply:
ComponentMap / ResourceMap are populated now while adding components/resources to the bundle (see `sync_components`), so both maps are ready when creating the systems that are using them. Both maps could possibly passed by reference immutably too, but i didnt found a solution for solving the lifetime problems i encountered. I sure would be happy if there is a solution for that if anyone can help with these.

Reply:
Ah, I didn't see that you had moved the logic for populating the map, that makes sense! I'll give it another look over and see if I can suggest how to fix the lifetime issues 👍